agenta-cli/sdk.py

```diff
@@ -0,0 +1,652 @@
+    ) -> Environment:
+        pass
+
+    def fetch(
+        self,
+        *,
+        environment_reference: Optional[Reference] = None,
+    ) -> Prompt:
+        pass
+
+    def commit(
+        self,
+        *,
+        environment_slug: Optional[str] = None,
+        variant_reference: Optional[Reference] = None,
+        variant: Optional[Variant] = None,
+        environment_reference: Optional[str] = None,
+        environment: Optional[Environment] = None,
+    ) -> Environment:
+        pass
+
+
+class PromptsClient:
+    def __init__(self):
+        pass
+
+    def create(
+        self,
+        slug: Optional[str] = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        *,
+        application_slug: Optional[str] = None,
+        template_key: Optional[str] = None,
+        prompt_reference: Optional[Reference] = None,
+        prompt: Optional[Prompt] = None,
+    ) -> Prompt:
+        pass
+
+    def fetch(
+        self,
+        *,
+        prompt_reference: Optional[Reference] = None,
+        environment_slug: Optional[Reference] = None,
+    ) -> Prompt:
+        pass
+
+    def commit(
+        self,
+        slug: Optional[str] = None,
+        parameters: Optional[Dict[str, Any]] = None,
+        *,
+        prompt_reference: Optional[Reference] = None,
+        prompt: Optional[Prompt] = None,
+    ) -> Prompt:
+        pass
+
+    def deploy(
+        self,
+        *,
+        prompt_reference: Optional[Reference] = None,
+        prompt: Optional[Variant] = None,
+        environment_slug: str,
+    ) -> Environment:
+        pass
+
+
+class Agenta:
+    def __init__(self):
+        self.applications = ApplicationsClient()
+        self.configurations = ConfigurationsClient()
+        self.variants = VariantsClient()
+        self.environments = EnvironmentsClient()
+        #
+        self.prompts = PromptsClient()
+
+
+ag = Agenta()
+
+
+########## APPLICATIONS ##########
+
+##### CREATE #####
+
+# 1.a. new application
+application: Application = ag.applications.create()
+
+# 1.b. new application
+application: Application = ag.applications.create(
+    application_slug="...",
+)
+
+##### FETCH #####
+
+# 1. existing configuration
+application: Application = ag.applications.fetch(
+    configuration_reference="...",
+)
+
+# 2. existing variant
+application: Application = ag.applications.fetch(
+    variant_reference="...",
+)
+
+# 3. existing environment
+application: Application = ag.applications.fetch(
+    environment_reference="...",
+)
+
+
+########## CONFIGURATIONS ##########
+
+##### CREATE #####
+
+# 1.a. new configuration
+configuration: Configuration = ag.configurations.create(
+    application_slug="...",
+    configuration_parameters="...",
+)
+
+# 1.b. new configuration
+configuration: Configuration = ag.configurations.create(
+    application_slug="...",
+    configuration_parameters="...",
+    configuration_slug="...",
+)
+
+# 2.a.i. existing configuration
```

Comment:
not sure what this does

Reply:
This only makes sense with the full extent of _Applications Structure_ where _Configurations_ are distinct from _Variants_. In this case, for this RFC, it would not be implemented.